pkg/scheduler/schedulercache/node_info.go

```diff
@@ -495,7 +494,7 @@ func (n *NodeInfo) FilterOutPods(pods []*v1.Pod) []*v1.Pod {
 
 // getPodKey returns the string key of a pod.
 func getPodKey(pod *v1.Pod) (string, error) {
-	return clientcache.MetaNamespaceKeyFunc(pod)
+	return string(pod.UID), nil
```

Comment:
Could you please add a check to return an error when pod.UID is empty?

Reply:
sure. I thought it is always non-empty? can I ask when it's empty?

Reply:
It shouldn't be empty, but a bug in the system could leave it empty. It could also be left empty in our tests. So, it would be good to check and return an error.

Reply:
make sense. just curious.